backend/infrahub/core/manager.py

```diff
@@ -603,7 +606,25 @@ async def get_one_by_hfid(
         if not node_schema.human_friendly_id or len(node_schema.human_friendly_id) != len(hfid):
             raise NodeNotFoundError(branch_name=branch.name, node_type=kind, identifier=hfid_str)
 
-        filters = {node_schema.human_friendly_id[idx]: item for idx, item in enumerate(hfid)}
+        # TODO: create a function like `hfid_to_graphql_filters` for this?
+        filters = {}
+        for idx, item in enumerate(hfid):
+            key = node_schema.human_friendly_id[idx]
+            path = node_schema.parse_schema_path(path=key, schema=registry.schema.get_schema_branch(name=branch.name))
+
```

Comment:
you could reduce the number of lines here if you want by moving the `python_type = path.attribute_schema.get_class().type` outside of the conditional
something like
```
if path.is_type_relationship:
    rel_schema = ...
    path = ...
 python_type = path.attribute_schema.get_class().type
```